### ISSUE TITLE
hwloc reader for resource doesn't fill in rank 

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-sched#453]
Just ran a simple test of the resource-matching service with RV1 emitter (I used `rv1_nosched` emitter instead of `rv1` more specifically). I will see if I can add rank support. But if rank support is already there but somehow this test doesn't appear to run incorrect, please let me know. 

```
ahn1@c7ec51045e3e:/usr/src/t$ flux resource match allocate data/resource/jobspecs/basics/test008.yaml
JOBID                STATUS               AT                   OVERHEAD (Secs)
0                    ALLOCATED            Now                  0.000555992126465
================================================================================
MATCHED RESOURCES:
{"execution":{"R_lite":[{"rank":-1,"node":"c7ec51045e3e","children":{"core":"3"}}]}}
```

**Comment by grondo** [comment]
>  But if rank support is already there but somehow this test doesn't appear to run incorrect, please let me know.

Sorry, I didn't parse that @dongahn.  How can I help?

The current execution system prototype does require correct `rank` information, and ignores any hostname in `node`.

**Comment by dongahn** [comment]
Sorry @grondo. I should have been clearer. The comment was more for @SteVwonder who added hwloc reader support within resource.  Notice:

```
{"R_lite":[{"rank":-1,
```

The emitter simply emits whatever value we have for the `rank` field within each resource vertex. The test shows that this field just has the initial value which is -1. So it seems I will need to add "rank" support for the hwloc reader. I just wanted to confirm with @SteVwonder that it is not that case that he already added rank support and my test somehow wasn't picking up.

**Comment by grondo** [comment]
Ah, ok, sorry!

**Comment by dongahn** [comment]
The most of rank support was already there. Apparently, some BGL algorithms call either the copy constructor or assignment constructor of a resource vertex, which doesn't copy the rank field:  https://github.com/dongahn/flux-sched/blob/match-format/resource/schema/resource_data.cpp#L42

An easy fix is to add rank support into those constructors, which I will do for now.

Then, still I need to fully understand which BGL algorithms are calling these methods. Currently, the code doesn't make a deep copy of the "schedule" data of a resource vertex, so depending on when and how these methods are called, the copying can having a side effect. I vaguely remember I actually tested this when I first implemented, this but it wouldn't hurt to verify this once again since it has been for a while.

In any case, at some point, adding copy construction of the schedule data also would be good so that we don't have remember safety properties like this once and for all. But to get there, I need a copy constructor (in C) for planner and planner_multi.
